airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/factory.py

```diff
@@ -238,3 +268,40 @@ def _is_builtin_type(cls) -> bool:
         if not cls:
             return False
         return cls.__module__ == "builtins"
+
+    @staticmethod
+    def _transform_interface_to_union(cls: type):
+        copy_cls = type(cls.__name__ + "Copy", cls.__bases__, dict(cls.__dict__))
+        class_fields = fields(copy_cls)
+        for field in class_fields:
+            unpacked_field_types = DeclarativeComponentFactory.unpack(field.type)
+            copy_cls.__annotations__[field.name] = unpacked_field_types
+
+    @staticmethod
+    def unpack(field_type: type):
```

Comment:
This is where it gets particularly fun. This whole recursive function is necessary to retain the existing structure of any declarative components that must be expanded, without losing the shape of the container they are in. We basically retrieve the underlying type and use that as the next recursive call. And once we have unpacked the underlying classes, we package it back into its original container. I added a handful of tests to verify this was where the subclass replacement became less than trivial